### PR TITLE
UCT/UCP/API: Add RKEY_PTR flag on component

### PR DESCRIPTION
**Pull request by brminich**
## What
Introduce `UCT_COMPONENT_FLAG_RKEY_PTR` and use it instead of `UCT_MD_FLAG_RKEY_PTR`

## Why ?
`uct_rkey_ptr()` function is invoked for component (not memory domain),  so the flag indicating support for this API needs to be checked/present on component rather than md

### CODE REVIEW

**Review by hoopoepg — APPROVED** [review]

File: src/uct/api/uct.h [hunk 1]
```diff
@@ -290,7 +290,13 @@ enum {
      * If set, the component supports @ref uct_cm_h functionality.
      * See @ref uct_cm_open for details.
      */
-    UCT_COMPONENT_FLAG_CM = UCS_BIT(0)
+    UCT_COMPONENT_FLAG_CM       = UCS_BIT(0),
+
+    /**
+     * Component supports direct access to remote memory via a pointer that is
+     * returned by @ref uct_rkey_ptr
```
Comment (yosefe) [inline src/uct/api/uct.h:297]: `If set, the component supports direct access to remote memory using a local pointer returned from @ref uct_rkey_ptr function`

File: src/ucp/core/ucp_context.h [hunk 2]
```diff
@@ -549,6 +549,14 @@ ucp_tl_iface_bandwidth(ucp_context_h context, const uct_ppn_bandwidth_t *bandwid
            (bandwidth->shared / context->config.est_num_ppn);
 }
 
+static UCS_F_ALWAYS_INLINE uct_component_attr_t*
```
Comment (yosefe) [inline src/ucp/core/ucp_context.h:552]: maybe return const pointer?

File: src/uct/api/uct.h [hunk 3]
```diff
@@ -2678,6 +2686,8 @@ ucs_status_t uct_rkey_unpack(uct_component_h component, const void *rkey_buffer,
  * This routine returns a local pointer to the remote memory
  * described by the rkey bundle. The MD must support
  * @ref UCT_MD_FLAG_RKEY_PTR flag.
+ * @note @ref UCT_MD_FLAG_RKEY_PTR is deprecated and replaced by the
+ * @ref UCT_COMPONENT_FLAG_RKEY_PTR component flag.
```
Comment (yosefe) [inline src/uct/api/uct.h:2690]: let's replace lines 2687-2688 instead of adding a note

**Review by yosefe — COMMENTED** [review]
shall we also update tl_info?

File: src/uct/api/uct.h (hunk 3 above)
Comment (brminich) [reply src/uct/api/uct.h:2690]: maybe better keep it  showing that backward compatibility is not broken?

Comment (yosefe) [reply src/uct/api/uct.h:2690]: i think the documentation should aim for new application and show the recommended/main way to use the API.
older application, that need the backward compat, are already written and don't need the API doc anymore.
if really important, we can add note that the component flag replaced the older MD flag.

Comment (brminich) [reply src/uct/api/uct.h]: ok, fixed

**Review by yosefe — DISMISSED** [review]

**Comment by yosefe** [comment]
@shamisp @tonycurtis can you pls review?

**Review by yosefe — APPROVED** [review]

**Review by tonycurtis — APPROVED** [review]

**Review by shamisp — APPROVED** [review]

**Pull request merged**